stl/inc/xutility

```diff
@@ -2497,16 +2509,22 @@ namespace ranges {
 
             template <class _Ty>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                _STL_INTERNAL_CHECK(is_lvalue_reference_v<_Ty>);
                 using _UnCV = remove_cvref_t<_Ty>;
+
                 if constexpr (is_array_v<_UnCV>) {
-                    return {_St::_Array, true};
+                    if constexpr (extent_v<_UnCV> != 0) {
+                        return {_St::_Array, true};
+                    } else {
+                        return {_St::_None};
+                    }

```

Comment:
Ditto `is_bounded_array_v` and `is_unbounded_array_v`

Reply:
Ditto "this is cheaper".